src/lib/pages/account-details/index.tsx

```diff
@@ -447,24 +438,26 @@ const AccountDetails = () => {
   const router = useRouter();
   const { isSomeValidAddress } = useValidateAddress();
 
-  const accountAddressParam = getFirstQueryParam(
-    router.query.accountAddress
-  ).toLowerCase() as Addr;
-  // TODO: fix assertion later
-  const tab = getFirstQueryParam(router.query.tab) as TabIndex;
+  const validated = zAccDetailQueryParams.safeParse(router.query);
 
   useEffect(() => {
-    if (router.isReady && tab) track(AmpEvent.TO_ACCOUNT_DETAIL, { tab });
-  }, [router.isReady, tab]);
+    if (router.isReady && validated.success && validated.data.tab)
+      track(AmpEvent.TO_ACCOUNT_DETAIL, { tab: validated.data.tab });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router.isReady]);
+
+  if (!validated.success) return <InvalidAccount />;
+
+  const { accountAddress, tab } = validated.data;
 
   return (
     <PageContainer>
-      {!isSomeValidAddress(accountAddressParam) ? (
+      {!isSomeValidAddress(accountAddress) ? (
         <InvalidAccount />
       ) : (
         <AccountDetailsBody
-          accountAddressParam={accountAddressParam}
-          tab={tab}
+          accountAddressParam={accountAddress}
+          tabParam={tab}
```

Comment:
change props to `accountAddress` and `tab` as they are not from query params any more, then we can just `...validated.data` here